src/components/GenderProvider/GenderProvider.js

```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+
+import GenderContext from '../../context/GenderContext';
+import { MALE } from '../../constants/genders';
+
+const GenderProvider = ({ children }) => {
+  const onGenderChange = ({ target }) =>
```

Comment:
I would go with something a bit more generic:
```
const handleGenderChange = (gender) => {
  setGender(gender);
}

const [gender, setGender] = useState(MALE);
```

Also it is good practice to keep only flat values when using **useState**. For objects there is **useReducer**.